src/main/kotlin/com/demonwav/mcdev/nbt/tags/TagByteArray.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.nbt.tags
+
+import java.io.DataOutputStream
+import java.util.Arrays
+
+class TagByteArray(override val value: ByteArray) : NbtValueTag<ByteArray>(ByteArray::class.java) {
+    override val payloadSize = 4 + value.size
+    override val typeId = NbtTypeId.BYTE_ARRAY
+
+    override fun write(stream: DataOutputStream) {
+        stream.writeInt(value.size)
+        for (byte in value) {
+            stream.writeByte(byte.toInt())
+        }
+    }
+
+    override fun toString() = toString(StringBuilder(), 0, WriterState.COMPOUND).toString()
+
+    override fun toString(sb: StringBuilder, indentLevel: Int, writerState: WriterState): StringBuilder {
+        sb.append("bytes(")
```

Comment:
The `joinTo` method can handle prefix/postfix for you: `value.joinTo(buffer = sb, prefix = "bytes(", separator = ", ", postfix = ")");`